pkg/registry/registry.go

```diff
@@ -302,6 +265,56 @@ func (r *registry) ReleaseManifests(layer v1.Layer) (string, string, error) {
 	return version, imageURL, nil
 }
 
+func (r *registry) GetLayersDigests(ctx context.Context, entry string) (string, []string, []crane.Option, error) {
+	registry, err := r.registryFromImageURL(entry)
+	if err != nil {
+		return "", nil, nil, err
+	}
+
+	auth, err := r.getImageRegistryCredentials(ctx, registry)
+	if err != nil {
+		return "", nil, nil, err
+	}
+
+	var repo string
+
+	if hash := strings.Split(entry, "@"); len(hash) > 1 {
+		repo = hash[0]
+	} else if tag := strings.Split(entry, ":"); len(tag) > 1 {
+		repo = tag[0]
+	}
```

Comment:
If `entry` does not contain neither `@` nor `:`, then `repo` will remain empty. Should we then set it to `entry`?
```suggestion
	repo := entry

	if hash := strings.Split(repo, "@"); len(hash) > 1 {
		repo = hash[0]
	} else if tag := strings.Split(repo, ":"); len(tag) > 1 {
		repo = tag[0]
	}
```

Reply:
i don't think so. we always assume that there is either a tag or a hash in the image. leaving the repo empty will result in failure to get the image, which i think is ok. Another option is to return an error in case repo is empty

Reply:
Returning an error sounds good to me.

Reply:
fixed